### ISSUE TITLE
🔴 Implement Proper Model Validation

### ISSUE THREAD

**Issue by gtlntw** [gtlntw/TradingBot-8888#4]
## Problem
Current model validation has serious flaws that could lead to production failures:
- **Data leakage**: Using future data to predict the past in train/test splits
- **Overfitting**: Models work on training data but fail in live trading
- **No drift detection**: Model performance degrades over time without notification
- **Invalid backtests**: Results don't reflect real-world trading conditions

## Root Cause Analysis
```python
# ❌ CURRENT PROBLEM: Regular split on time series
split_idx = int(len(df) * 0.8)  # This mixes time periods!
X_train = X[:split_idx]         # Past and future data mixed
X_test = X[split_idx:]          # Could be predicting past with future
```

This approach violates the fundamental rule: **never use future data to predict the past**.

### 1. Time-Series Cross-Validation
- [ ] **Implement TimeSeriesSplit**: Respect temporal order in validation
- [ ] **Walk-forward analysis**: Test on multiple sequential periods
- [ ] **Gap validation**: Add time gaps between train/test to simulate real delays
- [ ] **Purging**: Remove overlapping periods that could cause leakage

### 2. Model Drift Detection
- [ ] **Performance monitoring**: Track model accuracy over time
- [ ] **Statistical tests**: Detect distribution shifts in input features
- [ ] **Prediction confidence**: Monitor when model becomes uncertain
- [ ] **Automatic retraining**: Trigger retraining when drift detected

### 3. Production Validation Framework
- [ ] **Out-of-sample testing**: Hold out recent data for final validation
- [ ] **Live paper trading**: Test models with real-time data before live trading
- [ ] **A/B testing**: Compare new models against existing ones
- [ ] **Rollback mechanism**: Quickly revert to previous model if new one fails

### Time-Series Cross-Validation
```python
from sklearn.model_selection import TimeSeriesSplit

def time_series_validation(X, y, model, n_splits=5):
    tscv = TimeSeriesSplit(n_splits=n_splits)
    scores = []
    
    for train_idx, test_idx in tscv.split(X):
        # Ensure no data leakage
        X_train, X_test = X.iloc[train_idx], X.iloc[test_idx]
        y_train, y_test = y.iloc[train_idx], y.iloc[test_idx]
        
        # Train on past, predict future
        model.fit(X_train, y_train)
        score = model.score(X_test, y_test)
        scores.append(score)
    
    return scores
```

### Walk-Forward Analysis
```python
def walk_forward_validation(data, model, window_size=252, step_size=30):
    """
    Train on rolling window, test on next period
    window_size: 252 days (1 year of trading days)
    step_size: 30 days (monthly retraining)
    """
    results = []
    
    for i in range(window_size, len(data) - step_size, step_size):
        # Training window
        train_start = i - window_size
        train_end = i
        
        # Test window
        test_start = i
        test_end = i + step_size
        
        # Train and evaluate
        X_train = data[train_start:train_end]
        X_test = data[test_start:test_end]
        
        model.fit(X_train)
        score = model.evaluate(X_test)
        results.append(score)
    
    return results
```

### Model Drift Detection
```python
class ModelDriftDetector:
    def __init__(self, baseline_performance, alert_threshold=0.8):
        self.baseline = baseline_performance
        self.threshold = alert_threshold
        self.recent_scores = []
    
    def check_drift(self, recent_score):
        self.recent_scores.append(recent_score)
        
        # Keep only last 30 predictions
        if len(self.recent_scores) > 30:
            self.recent_scores.pop(0)
        
        avg_recent = np.mean(self.recent_scores)
        
        if avg_recent < self.baseline * self.threshold:
            return {
                'drift_detected': True,
                'baseline': self.baseline,
                'recent': avg_recent,
                'degradation': 1 - (avg_recent / self.baseline)
            }
        
        return {'drift_detected': False}
```

## Success Criteria
- [ ] **Zero data leakage**: All validation respects temporal order
- [ ] **Realistic performance**: Backtest results match live trading performance
- [ ] **Drift detection**: System alerts when model performance degrades >20%
- [ ] **Automated retraining**: Models retrain automatically when drift detected
- [ ] **Validation metrics**: Track validation scores across multiple time periods

## Files to Modify
- `trading_bot/models/trainer.py` (validation methods)
- `trading_bot/evaluation/validator.py` (new file)
- `trading_bot/models/drift_detector.py` (new file)
- `trading_bot/cli.py` (validation commands)

## Testing Plan
1. **Synthetic data test**: Create data with known patterns to verify no leakage
2. **Historical validation**: Apply to 2+ years of Bitcoin data
3. **Live paper trading**: 30-day validation period before live trading
4. **Performance tracking**: Monitor for 90 days post-deployment

## Priority
🔴 **Critical** - Essential for production trading reliability

## References
- [Advances in Financial Machine Learning](https://www.wiley.com/en-us/Advances+in+Financial+Machine+Learning-p-9781119482086) - Chapter 7: Cross-Validation
- [QuantStart: Walk-Forward Analysis](https://www.quantstart.com/articles/walk-forward-analysis/)
- [Model Drift Detection Techniques](https://towardsdatascience.com/concept-drift-detection-8a6f9b8a820f)
